### PR TITLE
EPMDJ-6132 [BE] [Admin] Fix tests for multiple isntance

### PR DESCRIPTION
**Pull request by Mazel-Tovr**

### CODE REVIEW

File: test-framework/src/main/kotlin/com/epam/drill/e2e/E2ETest.kt [hunk 1]
```diff
@@ -33,12 +33,95 @@ import kotlin.time.*
 
 
 abstract class E2ETest : AdminTest() {
-    private val agents = ConcurrentHashMap<AgentKey, AgentStruct>()
+
+    private val agents = ConcurrentHashMap<String, ConcurrentLinkedQueue<Pair<AgentKey, AgentStruct>>>()
```
Comment (LikhanovMaxim) [inline test-framework/src/main/kotlin/com/epam/drill/e2e/E2ETest.kt]: what is String? id or smth else? maybe it should use typealias

File: test-framework/src/main/kotlin/com/epam/drill/e2e/E2ETest.kt [hunk 2]
```diff
@@ -33,12 +33,95 @@ import kotlin.time.*
...
+
+    fun connectAgent(
+        ags: AgentWrap,
+        startClb: suspend () -> Unit = {},
+        bl: suspend AsyncTestAppEngine.(AdminUiChannels, Agent) -> Unit
+    ): E2ETest {
```
Comment (LikhanovMaxim) [inline test-framework/src/main/kotlin/com/epam/drill/e2e/E2ETest.kt]: ags, Clb, bl are not well naming

File: test-framework/src/main/kotlin/com/epam/drill/e2e/E2ETest.kt [hunk 3]
```diff
@@ -33,12 +33,95 @@ import kotlin.time.*
...
+                    )
+                )
+            )
+        } else {
+            agents[ags.id]?.add(AgentKey(ags.id, ags.instanceId) to AgentStruct(ags, startClb, bl, mutableListOf()))
+        }
+        return this
+    }
+
+    private suspend fun TestApplicationEngine.createWsConversation(
```
Comment (LikhanovMaxim) [inline test-framework/src/main/kotlin/com/epam/drill/e2e/E2ETest.kt]: this method should be lower than 'fun createSimpleAppWithUIConnection'

File: test-framework/src/main/kotlin/com/epam/drill/e2e/E2ETest.kt [hunk 4]
```diff
@@ -68,66 +151,36 @@ abstract class E2ETest : AdminTest() {
                     val globLaunch = application.launch(handler) {
                         watcher?.invoke(asyncEngine, glob)
                     }
-                    val cs = mutableMapOf<String, AdminUiChannels>()
+
+                    val cs = ConcurrentHashMap<String, AdminUiChannels>()
```
Comment (LikhanovMaxim) [inline test-framework/src/main/kotlin/com/epam/drill/e2e/E2ETest.kt]: naming - cs

File: test-framework/src/main/kotlin/com/epam/drill/e2e/E2ETest.kt [hunk 5]
```diff
@@ -68,66 +151,36 @@ abstract class E2ETest : AdminTest() {
...
-                                    )
-                                    while (globLaunch.isActive)
-                                        delay(100)
-
-                                }
-                                thens.forEach { (ain, it) ->
-                                    glob.receive()
+                                val currentInstance = queue.poll()
+                                val agentId = currentInstance.first.id
+                                if (cs[agentId] == null) {
```
Comment (LikhanovMaxim) [inline test-framework/src/main/kotlin/com/epam/drill/e2e/E2ETest.kt]:
```suggestion
                                cs[agentId]?.let {
```

File: tests/src/test/kotlin/e2e/MultipleInstanceProcessingTest.kt [hunk 6]
```diff
@@ -24,12 +24,14 @@ import kotlin.test.*
 
 class MultipleInstanceProcessingTest : E2ETest() {
 
+    private val agentName = "java-agent"
+
     @Test
     fun `agent can have multiple instances`() {
-        createSimpleAppWithUIConnection {
-            connectAgent(AgentWrap("myagent", "1", "0.1.1"), {}) { ui, agent ->
+        createSimpleAppWithUIConnection() {
```
Comment (LikhanovMaxim) [inline tests/src/test/kotlin/e2e/MultipleInstanceProcessingTest.kt]:
```suggestion
        createSimpleAppWithUIConnection {
```

File: tests/src/test/kotlin/e2e/MultipleInstanceProcessingTest.kt [hunk 7]
```diff
@@ -38,17 +40,57 @@ class MultipleInstanceProcessingTest : E2ETest() {
...
+    }
 
-                    agent1.`get-set-packages-prefixes`()
-                    agent1.`get-load-classes-datas`()
-                    val agentInfo1 = ui1.getAgent()
-                    agentInfo1?.status shouldBe AgentStatus.ONLINE
-                    agentInfo1?.instanceIds shouldBe setOf("1", "2")
+    @Test
+    fun `agent should be busy after new instance connect`() {
+        createSimpleAppWithUIConnection() {
```
Comment (LikhanovMaxim) [inline tests/src/test/kotlin/e2e/MultipleInstanceProcessingTest.kt]:
```suggestion
        createSimpleAppWithUIConnection {
```

**Review by LikhanovMaxim — CHANGES_REQUESTED** [review]

File: test-framework/src/main/kotlin/com/epam/drill/e2e/E2ETest.kt (hunk 5 above)
Comment (Mazel-Tovr) [reply test-framework/src/main/kotlin/com/epam/drill/e2e/E2ETest.kt]: cs[agentId]?.let -> if (cs[agentId] != null) 
but we need this block whent cs[agentId] == null

**Review by LikhanovMaxim — APPROVED** [review]

**Pull request merged**